src/templates/aws/advanced.ts

```diff
@@ -112,51 +111,190 @@ export default class Advanced {
       app_port                    = var.app_port
       private_subnets_cidr_blocks = module.vpc.private_subnets_cidr_blocks
-    
+
       nimble_office_ip = var.nimble_office_ip
     }
     `
 
-    injectToFile('main.tf', ssmModuleContent, this.options, {insertAfter: '# Security groups'})
+    injectToFile('main.tf', ssmModuleContent, this.options, {
+      insertAfter: '# Security groups',
+    })
+  }
+
+  private applyEcr(): void {
+    copyDir('aws/modules/ecr', 'modules/ecr', this.options)
+
+    const ecrVariablesContent = dedent`
+    variable "image_limit" {
+      description = "Sets max amount of the latest develop images to be kept"
+      type        = number
+    }\n\n`
+    appendToFile('variables.tf', ecrVariablesContent, this.options)
+
+    const ecrModuleContent = dedent`
+    module "ecr" {
+      source = "./ecr"
+
+      namespace   = var.app_name
+      owner       = var.owner
+      image_limit = var.image_limit
+    }
+    `
+
+    injectToFile('main.tf', ecrModuleContent, this.options, {
+      insertAfter: '# ECR',
+    })
+  }
+
+  private applyLog(): void {
+    copyDir('aws/modules/log', 'modules/log', this.options)
+
+    const logModuleContent = dedent`
+    module "log" {
+      source = "./modules/log"
+
+      namespace = var.app_name
+    }
+    `
+
+    injectToFile('main.tf', logModuleContent, this.options, {
+      insertAfter: '# Log',
+    })
+  }
+
+  private applyAlb(): void {
+    copyDir('aws/modules/alb', 'modules/alb', this.options)
+
+    const albVariablesContent = dedent`
+    variable "health_check_path" {
+      description = "Application health check path"
+      type = string
+    }
+
+    variable "domain" {
+      description = "Application domain"
+      type        = string
+    }\n\n`
+    appendToFile('variables.tf', albVariablesContent, this.options)
+
+    const albModuleContent = dedent`
+    module "alb" {
+      source = "./modules/alb"
+
+      vpc_id             = module.vpc.vpc_id
+      namespace          = var.app_name
+      app_port           = var.app_port
+      subnet_ids         = module.vpc.public_subnet_ids
+      security_group_ids = module.security_group.alb_security_group_ids
+      domain  = var.domain
+      health_check_path  = var.health_check_path
+    }
+    `
+
+    injectToFile('main.tf', albModuleContent, this.options, {
+      insertAfter: '# ALB',
+    })
+  }
+
+  private applyRds(): void {
+    copyDir('aws/modules/rds', 'modules/rds', this.options)
+
+    const rdsVariablesContent = dedent`
+    variable "rds_instance_type" {
+      description = "The RDB instance type"
+      type        = string
+    }
+
+    variable "rds_database_name" {
+      description = "RDS database name"
+      type = string
+    }
+
+    variable "rds_username" {
+      description = "RDS username"
+      type = string
+    }
+
+    variable "rds_password" {
+      description = "RDS password"
+      type = string
+    }
```

Comment:
please remove these variables from `applySsm()`, otherwise they will be duplicate

Reply:
Thanks, good catch!